### DISCUSSION TITLE
Override Button pointerover state to not set color

### DISCUSSION THREAD

**Discussion by josephnarai** [AvaloniaUI/Avalonia#6721] [Q&A] [answered]
Hi all, thanks again for the great community support!

We have a Button whose base color is being set in code (Background = Color).

We are setting the border color on hover via the Selector property, but we don't want to change the Background color we have set in code. My understadning is that in overriding the Button style, we can set new values for the colors, but if we don't set a color in our overridden style, then the default color will be used (this is what we observe too!) ....

So, my question is, is there a way to prevent the pointerover Selector from acutally changing the color, other than making a compleley new Style and applying that to the Button? We want to use this as the default style...

**Comment by MarkusKgit** [comment, marked as answer]
If I understood you correctly, here is how it could work:
```xaml
<Grid>
  <Grid.Styles>
    <Style Selector="Button:pointerover /template/ ContentPresenter#PART_ContentPresenter">
      <Setter Property="Background" Value="{Binding $parent[Button].Background} "/>
    </Style>
  </Grid.Styles>
  <Button Name="RedButton" Content="Red Button" />
</Grid>
```
```csharp
var button = this.FindControl<Button>("RedButton");
button.Background = Brushes.Red;
```
![Button](https://user-images.githubusercontent.com/13160892/137264385-c0b7ecd7-11b2-4ccf-86e0-6ca94a221579.gif)

**Reply by josephnarai** [reply]
Awesome, thanks!

We ended up just making our own subclass of Button, along with our own style. Then it didn't fall back to the default style setting.
